### PR TITLE
[RW-12298][risk=low] upgrade Terra docker to 2.2.12

### PR DESCRIPTION
**Pull request by yonghaoy**
This includes Hail upgrade and bioconductor upgrade.
Need to wait after Terra release to merge to prod: https://github.com/DataBiosphere/leonardo/pull/4566

---
**PR checklist**

- [ ] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [ ] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: api/config/config_prod.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
     "xAppIdValue": "AoU-RW",
     "timeoutInSeconds": 40,
     "lenientTimeoutInSeconds": 180,
-    "jupyterDockerImage": "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.7",
```
Comment (Qi77Qi) [inline api/config/config_prod.json:15]: shall we separate prod in a follow up just in case we found anything surprising?

**Review by Qi77Qi — APPROVED** [review]

File: api/config/config_prod.json (hunk 1 above)
Comment (yonghaoy) [reply api/config/config_prod.json]: good idea. Reverting preprod and prod

File: api/config/config_stable.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
     "xAppIdValue": "stable-AoU-RW",
     "timeoutInSeconds": 40,
     "lenientTimeoutInSeconds": 180,
-    "jupyterDockerImage": "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.7",
+    "jupyterDockerImage": "us.gcr.io/broad-dsp-gcr-public/terra-jupyter-aou:2.2.12",
```
Comment (jmthibault79) [inline api/config/config_stable.json:15]: stable and staging also use Terra Prod, so they will also need to wait for the Terra release

Comment (yonghaoy) [reply api/config/config_stable.json:15]: Leo PR is merged, most likely the change would be released to prod tomorrow or some time this week. And we already cut release for this week. Most likely it should be fine and staging & stable is actively used. I might will just have staging /stable in this pr and do preprod and prod in seperate PR as long as we verify it works in other envs.

**Pull request merged**